accepted/future-releases/type-modifiers/feature-specification.md

```diff
@@ -447,114 +449,108 @@ class UsesAsSuperclass extends Both {}
+Many combinations don't make sense:
 
-*   `sealed` implies `abstract`, so they can't be combined.
-*   `sealed` implies non-extensibility, so can't be combined with `interface`
-    or `final`.
-*   `sealed` implies non-implementability, so can't be combined with `base` or
-    `final`.
 *   `base`, `interface`, and `final` all control the same two capabilities so
     are mutually exclusive.
-*   `mixin` as a modifier can obviously only be applied to a class. It can be
-    combined with any other modifiers that can be applied to a class.
-
-The remaining valid combinations are:
+*   `sealed` types can't be constructed so it's redundant to combine with
+    `abstract`.
+*   `sealed` types can't be extended or implemented, so it's redundant to
+    combine with `final`.
+*   `sealed` types can't be extended so it contradicts `base`.
+*   `sealed` types can't be implemented, so it contradicts `interface`.
+*   `sealed` types can't be mixed in outside of their library, so it contradicts
+    `mixin` on a class. *It's useful to allow `sealed` on a mixin declaration
+    because the mixin can be applied within the same library. But a class can
+    already be used as a mixin within its own library even without the `mixin`
+    modifier, so allowing `sealed mixin class` adds nothing.*
+*   `interface` and `final` classes prevent the class from being used as a
+    superclass but mixing in a mixin class also makes the class a superclass, so
+    they contradict the `mixin` modifier. *An `interface mixin class M {}` would
+    prohibited from appearing in an `extends` clause but could still be in
+    `extends Object with M` which has the exact same effect.*
+*   `mixin` as a modifier can obviously only be applied to a class.
+*   Mixin declarations can't be constructed, so `abstract` is redundant.
+
+The remaining valid combinations and their capabilities are:
+
+| Declaration | Construct? | Extend? | Implement? | Mix in? | Exhaustive? |
+|--|--|--|--|--|--|
+|`class`                    |**Yes**|**Yes**|**Yes**|No     |No     |
+|`base class`               |**Yes**|**Yes**|No     |No     |No     |
+|`interface class`          |**Yes**|No     |**Yes**|No     |No     |
+|`final class`              |**Yes**|No     |No     |No     |No     |
+|`sealed class`             |No     |No     |No     |No     |**Yes**|
+|`abstract class`           |No     |**Yes**|**Yes**|No     |No     |
+|`abstract base class`      |No     |**Yes**|No     |No     |No     |
+|`abstract interface class` |No     |No     |**Yes**|No     |No     |
+|`abstract final class`     |No     |No     |No     |No     |No     |
+|`mixin class`              |**Yes**|**Yes**|**Yes**|**Yes**|No     |
+|`base mixin class`         |**Yes**|**Yes**|No     |**Yes**|No     |
+|`abstract mixin class`     |No     |**Yes**|**Yes**|**Yes**|No     |
+|`abstract base mixin class`|No     |**Yes**|No     |**Yes**|No     |
+|`mixin`                    |No     |No     |**Yes**|**Yes**|No     |
```

Comment:
This is pretty much my proposal at the bottom of [this comment](https://github.com/dart-lang/language/issues/2723#issuecomment-1354093842).  If we allow extending a normal `mixin`, then the two `abstract ... mixin class` combinations also become redundant.

Reply:
Yup! I wrote a comment on that issue comparing your proposal and Lasse's with this one. I think we're converging modulo a couple of questions about mixins.